### ISSUE TITLE
Change UserProfile.is_staff to be True for all mozilla.com emails

### ISSUE THREAD

**Issue by diox** [mozilla/addons-server#15601]
`is_staff` is a `UserProfile` property django uses to gate login to the django admin. It doesn't give any particular access, if you don't have any permissions but that property returns `True`, then you can log in to the admin but you'd just see an empty admin (provided you can even reach the page in the first place, since we have nginx rules that force you to be part of a specific VPN group for the admin to work).

Typically it's implemented as a column in the database, but in our case it's a `@property` returning`acl.action_allowed_user(self, amo.permissions.ANY_ADMIN)` - essentially returning `True` as long as the user has any `Admin:<Something>` permission.

I'd like to change it to be `True` for all mozilla employees - anyone with a `mozilla.com` email. We'd then be able to use waffle flag's `staff` property, which enables a given flag for all `staff` members, and be able to enable a flag for all mozilla employees.

**Comment by wagnerand** [comment]
Can we make sure we're not giving actual access to sensitive or privileged resources or information to all employees?

I am concerned that it will no longer be obvious/visible from the groups/permissions definitions if a user has special privileges, which is something that admin need to check regularly as well as on demand.

**Comment by diox** [comment]
The docstring is already pretty clear, and I've checked: neither django nor we are using it anywhere else.

**Comment by Sparsh-Bansal** [comment]
@diox  As changing `is_staff=True` for all the employees is not giving any access to the admin page .So, should I change the `is_staff` property to always return True and land a PR on it ?

**Comment by diox** [comment]
No, that would defeat the purpose. This issue is already assigned to me for a reason, a lot of context and understanding of the project is required to work on it.

**Comment by diox** [comment]
One problem I had with this is that it would prevent contributors from working on the admin on their local envs as they wouldn't be able to log in to a mozilla.com account. https://github.com/mozilla/addons-server/issues/15701 solved this nicely, local environments now bypass FxA by default, allowing you to log in with any email.

**Comment by AlexandraMoga** [comment]
@diox I've created a user with a `mozilla.com` address and without setting up any permissions I've tried to access:

- https://addons-internal.dev.mozaws.net/en-US/admin/models/ (with VPN on) => You don't have permission to view or edit anything.
- https://reviewers.addons-dev.allizom.org/en-US/reviewers/ or https://reviewers.addons-dev.allizom.org/en-US/reviewers/review/13431-3-listed-v2-unlisted => Not allowed to access the page

So, the user doesn't seem to have access to any permission guarded content. 

Let me know if there is anything more specific to be tested.
